app/javascript/components/Schedule/TimeSlotModal.js

```diff
@@ -0,0 +1,173 @@
+import React, { Component } from "react"
+import PropTypes from "prop-types"
+import { patchTimeSlot } from '../../apiCalls';
+
+class TimeSlotModal extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      sessionSelected: this.props.matchedSession 
+      ? this.props.matchedSession
+      : ''
+    }
+  }
+
+  componentDidMount() {
+    const { csrf, slot, matchedSession } = this.props
+  }
+
+  changeSession = (e) => {
+    let session = this.props.sessions.find(s => s.title === e.target.value )
+    this.setState({ sessionSelected: session })
+  }
+
+  close = () => {
+    this.props.closeModal()
+  }
+
+  saveChanges = () => {
+    const { sessionSelected } = this.state
+    const { csrf, closeModal, handleMoveSessionResponse, title, track, presenter, description } = this.props
+    
+    let slot
+    if (!sessionSelected) {
+      slot = Object.assign(this.props.slot, {title, track_id: track, presenter, description})
+    } else {
+      slot = this.props.slot
+    }
+    
+    patchTimeSlot(slot, (sessionSelected || null), csrf)
+      .then(response => response.json())
+      .then(data => {
+        const { sessions, slots, unscheduled_sessions } = data
+        handleMoveSessionResponse(sessions, unscheduled_sessions, slots)
+        closeModal()
+      })
+  }
+
+  render() {
+    const { 
+      slot, 
+      matchedSession, 
+      unscheduledSessions, 
+      tracks, 
+      sessionFormats, 
+      title, 
+      track, 
+      presenter, 
+      description, 
+      updateSlot 
+    } = this.props
+
+    const { sessionSelected } = this.state
+
+    let sessionOptions
+    if (sessionSelected) {
```

Comment:
![Screen Shot 2019-06-12 at 9 56 51 AM](https://user-images.githubusercontent.com/34146574/59362416-062f0180-8cf9-11e9-837a-b0d5fc31f113.png)
If you choose a program session by accident there no way to clear it without closing the modal.

Reply:
Good point, I can change that!

Reply:
Everything else looks great!